src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -249,26 +252,50 @@ public bool RecoverDatabase(TransactionHeader* txHeader, Action<string> addToIni
 
                 long minPageChecked = -1;
 
-                // we need to iterate from the end in order to filter out pages that was overwritten by later transaction
-
-                for (var i = sortedPages.Length - 1; i >= 0; i--)
+                if (_env.Options.SkipChecksumValidationOnDatabaseLoading == false)
                 {
-                    using (tempTx) // release any resources, we just wanted to validate things
+                    // we need to iterate from the end in order to filter out pages that was overwritten by later transaction
+                    addToInitLog?.Invoke($"Validate checksum on {sortedPages.Length} pages");
+                    var sp = Stopwatch.StartNew();
+                    for (var i = sortedPages.Length - 1; i >= 0; i--)
                     {
-                        var modifiedPage = sortedPages[i];
+                        if (sp.Elapsed.TotalSeconds >= 60)
+                        {
+                            sp.Restart();
+                            addToInitLog?.Invoke($"Still calculating checksum.. ({sortedPages.Length - i} out of {sortedPages.Length}");
+                        }
 
-                        var ptr = (PageHeader*)_dataPager.AcquirePagePointerWithOverflowHandling(tempTx, modifiedPage, null);
+                        using (tempTx) // release any resources, we just wanted to validate things
+                        {
+                            var modifiedPage = sortedPages[i];
 
-                        var maxPageRange = modifiedPage + VirtualPagerLegacyExtensions.GetNumberOfPages(ptr) - 1;
+                            var ptr = (PageHeader*)_dataPager.AcquirePagePointerWithOverflowHandling(tempTx, modifiedPage, null);
 
-                        if (minPageChecked != -1 && maxPageRange >= minPageChecked)
-                        {
-                            continue;
-                        }
+                            var maxPageRange = modifiedPage + VirtualPagerLegacyExtensions.GetNumberOfPages(ptr) - 1;
+
+                            if (minPageChecked != -1 && maxPageRange >= minPageChecked)
+                            {
+                                continue;
+                            }
 
-                        _env.ValidatePageChecksum(modifiedPage, ptr);
+                            _env.ValidatePageChecksum(modifiedPage, ptr);
 
-                        minPageChecked = modifiedPage;
+                            minPageChecked = modifiedPage;
+                        }
+                    }
+
+                    addToInitLog?.Invoke($"Validate completed");
+                }
+                else
+                {
+                    if (RuntimeInformation.OSArchitecture == Architecture.Arm || RuntimeInformation.OSArchitecture == Architecture.Arm64)
```

Comment:
Why do we have this limitation? If we are running on low end x86 machine, we might want this as well